cryptoservice/certificate.go

```diff
@@ -44,7 +24,7 @@ func GenerateCertificate(rootKey data.PrivateKey, gun string) (*x509.Certificate
 		return nil, fmt.Errorf("failed to create the certificate template for: %s (%v)", gun, err)
 	}
 
-	derBytes, err := x509.CreateCertificate(rand.Reader, template, template, publicKey, privateKey)
+	derBytes, err := x509.CreateCertificate(rand.Reader, template, template, rootKey.CryptoSigner().Public(), rootKey.CryptoSigner())
```

Comment:
In this instance we want the `ecdsa.PrivateKey` or `rsa.PrivateKey`, not the public key bytes as serialized for TUF, which is what `rootKey.Public()` would return.